CMakeLists.txt

```diff
@@ -154,6 +156,63 @@ set(ZLIB_SRCS
     uncompr.c
     zutil.c)
 
+
+# IBM s390 extensions
+if(DFLTCC OR CRC32VX)
+# Test to see if we can use the optimizations for S390
+CHECK_C_SOURCE_COMPILES("
+	#ifndef __s390x__
+		#error \"Target is not S390X\"
+	#endif
+	int main() { return 0; }
+	" HAS_S390X_SUPPORT)
+
+# check if we have secure_getenv
+set(CMAKE_REQUIRED_FLAGS -D_GNU_SOURCE=1)
+CHECK_C_SOURCE_COMPILES("
+	#include <stdlib.h>
+	int main() { char* _foo = secure_getenv(\"PWD\");return 0; }
+	" HAS_SECURE_GETENV)
+if(HAS_SECURE_GETENV)
+	add_definitions(-DZ_HAVE_SECURE_GETENV)
```

Comment:
$<$<BOOL:HAS_SECURE_GETENV>:Z_HAVE_SECURE_GETENV> in target_compile_definitions

same for the sources